patches/tModLoader/Terraria.ModLoader/CommandManager.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Terraria.ModLoader
+{
+	public static class CommandManager
+	{
+		internal static readonly IDictionary<string, List<ModCommand>> ChatCommands = new Dictionary<string, List<ModCommand>>();
+		internal static readonly IDictionary<string, List<ModCommand>> ServerCommands = new Dictionary<string, List<ModCommand>>();
+
+		internal static List<ModCommand> GetCommandList(CommandType type, string cmd)
+		{
+			List<ModCommand> cmdList;
+			if (type == CommandType.Chat)
+				ChatCommands.TryGetValue(cmd, out cmdList);
+			else
+				ServerCommands.TryGetValue(cmd, out cmdList);
+			return cmdList;
+		}
+
+		internal static void Add(ModCommand cmd)
+		{
+			var cmdList = GetCommandList(cmd.Type, cmd.Command);
+			if (cmdList != null)
+				cmdList.Add(cmd);
+			else if (cmd.Type == CommandType.Chat)
+				ChatCommands.Add(cmd.Command, new List<ModCommand>() {cmd});
+			else
+				ServerCommands.Add(cmd.Command, new List<ModCommand>() {cmd});
+		}
+
+		internal static void Unload()
+		{
+			ChatCommands.Clear();
+			ServerCommands.Clear();
+		}
+
+		internal static string[] ParseCommand(string text) => text.Substring(1).Split(' ');
+
+		internal static void PrintUsage(ModCommand mc) => Main.NewText("Usage: " + mc.Usage);
+
+		internal static void ProcessInput(string input, CommandType type, ref bool show)
+		{
+			var cmd = ParseCommand(input);
+			var cmdList = GetCommandList(type, cmd[0]);
+			if (cmdList == null) return;
+			var args = cmd.Skip(1).ToArray();
+			var mc = cmdList[0];
+
+			show = mc.Show;
+			if (cmdList.Count > 1)
+				Main.NewText("Error: Multiple definitions of command /" + mc.Command);
+			else if (!mc.VerifyArguments(args))
+				PrintUsage(mc);
```

Comment:
I figured if you were using a command that was designed to be hidden from chat you probably didn't want everyone seeing it if you messed up the syntax or something.  Also if there's multiple definitions I don't think it really matters either way, but if you want to change it that's fine with me.